### PR TITLE
Implement a boot pipeline unit for checking Pipfile hash

### PR DESCRIPTION
**Pull request by fridex**
... so that the recommendation engine checks consistency in the Pipenv files submitted.

## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/thamos/issues/617

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: thoth/adviser/boots/pipfile_hash.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""A boot that checks for Pipfile hash and reports any mismatch to users.."""
```
Comment (harshad16) [inline thoth/adviser/boots/pipfile_hash.py:18]:
```suggestion
"""A boot that checks for Pipfile hash and reports any mismatch to users."""
```

File: thoth/adviser/boots/pipfile_hash.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+if TYPE_CHECKING:
+    from ..pipeline_builder import PipelineBuilderContext
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@attr.s(slots=True)
+class PipfileHashBoot(Boot):
+    """A boot that checks for Pipfile hash and reports any mismatch to users.."""
```
Comment (harshad16) [inline thoth/adviser/boots/pipfile_hash.py:40]:
```suggestion
    """A boot that checks for Pipfile hash and reports any mismatch to users."""
```

**Review by harshad16 — APPROVED** [review]
lgtm,
thanks for the addition

File: thoth/adviser/boots/pipfile_hash.py (hunk 1 above)
Comment (fridex) [reply thoth/adviser/boots/pipfile_hash.py:18]: Nice catch, thanks! :+1:

**Comment by harshad16** [comment]
The changes looks good to me, thanks for all test update.
i think the pydocs change which i had suggested gotten overriden.

**Comment by fridex** [comment]
> The changes looks good to me, thanks for all test update.
> i think the pydocs change which i had suggested gotten overriden.

Cool, thanks for the review. :+1:

**Pull request merged**